src/Symmetry/Crystal.jl

```diff
@@ -506,3 +513,81 @@ function validate(cryst::Crystal)
 
     # TODO: Check that space group is closed and that symops have inverse?
 end
+
+#= Definitions of common crystals =#
+
+function cubic_crystal(; a=1.0)
+    lat_vecs = lattice_vectors(a, a, a, 90, 90, 90)
+    basis_vecs = [0, 0, 0]
+    Crystal(lat_vecs, basis_vecs)
+end
+
+function fcc_crystal(; a=1.0)
+    lat_vecs = lattice_vectors(a, a, a, 90, 90, 90)
+    basis_vecs = [[0, 0, 0]/2,
+                  [1, 1, 0]/2,
+                  [1, 0, 1]/2,
+                  [0, 1, 1]/2]
+    cryst = Crystal(lat_vecs, basis_vecs)
+    sort_sites!(cryst)
+    cryst
+end
+
+function fcc_primitive_crystal(; a=1.0)
+    lat_vecs = [1 1 0; 1 0 1; 0 1 1]' * a/2
+    basis_vecs = [[0, 0, 0]]
+    Crystal(lat_vecs, basis_vecs)
+end
+
+function bcc_crystal(; a=1.0)
+    lat_vecs = lattice_vectors(a, a, a, 90, 90, 90)
+    basis_vecs = [[0, 0, 0]/2,
+                  [1, 1, 1]/2,]
+    Crystal(lat_vecs, basis_vecs)
+end
+
+function bcc_primitive_crystal(; a=1.0)
+    lat_vecs = [1 1 -1; 1 -1 1; -1 1 1]' * a/2
+    basis_vecs = [[0, 0, 0]]
+    Crystal(lat_vecs, basis_vecs)
+end
+
+
+function diamond_crystal(; a=1.0)
+    lat_vecs = lattice_vectors(a, a, a, 90, 90, 90)
+    basis_vecs = [
+        [0, 0, 0]/4,
+        [2, 2, 0]/4,
+        [1, 1, 1]/4,
+        [3, 3, 1]/4,
+        [2, 0, 2]/4,
+        [0, 2, 2]/4,
+        [3, 1, 3]/4,
+        [1, 3, 3]/4,
```

Comment:
I notice the ordering changed. Is this using the Sunny sorting now?

Reply:
Yeah, we had previously discussed this briefly. There was some mixup between the vectors chosen by us manually, and by Spglib when we ask to construct all sites from the symmetry operations. I just changed these to agree with Spglib's list.

Reply:
Ah, right.